src/iceberg_destination.rs

```diff
@@ -0,0 +1,201 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use arrow::array::RecordBatch;
+use arrow_schema::{Field, Schema, SchemaRef};
+use futures::{pin_mut, StreamExt, TryStream};
+use iceberg::io::FileIO;
+use iceberg::spec::{
+    DataContentType, DataFileFormat, FormatVersion, Manifest, ManifestContentType, ManifestEntry,
+    ManifestFile, ManifestListWriter, ManifestMetadata, ManifestStatus, ManifestWriter, Operation,
+    PartitionSpec, Snapshot, Struct, Summary, TableMetadata, TableMetadataBuilder,
+};
+use iceberg::writer::file_writer::location_generator::{
+    DefaultFileNameGenerator, DefaultLocationGenerator,
+};
+use iceberg::writer::file_writer::{FileWriter, FileWriterBuilder, ParquetWriterBuilder};
+use iceberg::TableCreation;
+use log::info;
+use parquet::arrow::PARQUET_FIELD_ID_META_KEY;
+use parquet::file::properties::WriterProperties;
+use url::Url;
+use uuid::Uuid;
+
+use crate::error::DataLoadingError;
+
+pub async fn record_batches_to_iceberg(
```

Comment:
This function is almost 200 lines long and handle multiple tasks. You can break it into something like: 

```
let file_io = setup_file_io(...)
let iceberg_schema = assign_field_ids(...)
let metadata_v0 = prepare_metadata(...)
let data_files = write_record_batches(...)
let manifest_file = write_manifest(...)
write_snapshot(manifest_file, metadata_v0, target_url, file_io)
```

Besides, some values repeat along it, so you can store them in constants: 

```
const DEFAULT_SCHEMA_ID: i32 = 0;
const METADATA_PATH: &str = "metadata";
const PARTITION_FILENAME: &str = "part";
```

Reply:
Ok, I've moved out the more self-contained logic out of `record_batches_to_iceberg` now. The remaining logic could be split up further, but right now I have low confidence that we know the right way to do it. It will become clearer in future as we learn more about Iceberg and which iceberg-related features we want to build